invenio_app_rdm/upgrade_scripts/migrate_13_0_to_14_0.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# Invenio-App-RDM is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Record migration script from InvenioRDM 13.0 to 14.0.
+
+Disclaimer: This script is intended to be executed *only once*, namely when
+upgrading from InvenioRDM 13.0 to 14.0!
+If this script is executed at any other time, probably the best case scenario
+is that nothing happens!
+
+
+This script has been tested with following data:
+
+- record
+    - internal_notes
+"""
+
+from click import secho
+from invenio_access.permissions import system_identity
+from invenio_db import db
+from invenio_drafts_resources.services.records.errors import DraftNotCreatedError
+from invenio_rdm_records.proxies import current_rdm_records_service as records_service
+from invenio_search.engine import dsl
+
+
+def run_upgrade(has, migrate_record, migrate_draft):
+    """Run upgrade."""
+    # Handle published records
+    published_records = records_service.scan(
+        system_identity,
+        params={"allversions": True, "include_deleted": True},
```

Comment:
I am not 100% sure that the service layer edit/publish will work for deleted records.

Reply:
This question was not answered.